### DISCUSSION TITLE
Why do we need `System.Data.SqlTypes` in .NET Core?

### DISCUSSION THREAD

**Discussion by B1Z0N** [dotnet/runtime#67484] [Q&A] [answered]
1. From what I know it focuses solely on `Microsoft SQL Server` support types. And it's seems weird to me, why do we need this particular SQL in here, why omit `PostgreSQL`, `MySQL`, etc? I understand that it's microsoft and all things, but why adding this to standard library and not just as a side dependency?

2. And even if we take it as granted than it doesn't include any types to actually connect to the database, for that we need to install extra [`System.Data.SqlClient`](https://www.nuget.org/packages/System.Data.SqlClient) package from nuget.

I tried reading code and docs on it and haven't managed to find the answers. So what is the purpose?

**Comment by teo-tsirpanis** [comment, marked as answer]
These types are in the BCL mostly for historic reasons; they have been there since the beginning of .NET Framework. If it were redesigned today, perhaps they would have been in a separate package.

Which is the reason that `System.Data.SqlClient` which is bundled on .NET Framework was replaced by an out-of-band NuGet package called `Microsoft.Data.SqlClient`.

**Comment by MaceWindu** [comment]
Note that they also used by `System.Data.SqlServerCe`. While it is not .net core assembly, it works in .net core just fine (as there is no other option)
